internal/client/gitlab.go

```diff
@@ -98,6 +98,28 @@ func (c *gitlabClient) getDefaultBranch(_ *context.Context, repo Repo) (string,
 	return p.DefaultBranch, nil
 }
 
+// checkBranchExists checks if a branch exists
+func (c *gitlabClient) checkBranchExists(_ *context.Context, repo Repo, branch string) (bool, error) {
+	projectID := repo.Name
+	if repo.Owner != "" {
+		projectID = repo.Owner + "/" + projectID
+	}
+
+	// Verify if branch exists
+	_, res, err := c.client.Branches.GetBranch(projectID, branch)
+	if err != nil && res.StatusCode != 404 {
+		log.WithError(err).
+			Error("error verify branch existence")
+		return false, err
+	}
+
+	if res.StatusCode == 404 {
+		return false, nil
+	} else {
+		return true, nil
+	}
```

Comment:
```suggestion
	return res.StatusCode != 404 
```

Reply:
Are you sure about this one? Because the function has `(bool, error)` as return type. Should this be:

```
return res.StatusCode != 404, nil
```

Reply:
yes, forgot to add the `, nil`